src/solvers/memory_solution_history.C

```diff
@@ -242,15 +242,33 @@ void MemorySolutionHistory::retrieve(bool is_adjoint_solve, Real time)
   // Of course, we will *always* have to get the actual solution
   std::string _solution("_solution");
   *(_system.solution) = *(saved_vectors[_solution]);
+
+  // We need to call update to put system in a consistent state
+  // with the solution that was read in
+  _system.update();
```

Comment:
How did this not bite us before?

Reply:
We were doing the update when we would manually compute the sensitivity:
void HeatSystem::perturb_accumulate_residuals(ParameterVector & parameters_in)
{
  this->update();

Reply:
Ah, got it.  Okay, the update here makes sense, thanks.